src/vos/vos_pool.c

```diff
@@ -583,14 +583,17 @@ vos_pmemobj_create(const char *path, uuid_t pool_id, const char *layout,
 		goto umem_create;
 
 	if (!scm_sz) {
-		struct stat lstat;
+		if (!meta_sz) {
+			struct stat lstat;
 
-		rc = stat(path, &lstat);
-		if (rc != 0)
-			return daos_errno2der(errno);
-		meta_sz = lstat.st_size;
-	} else if (meta_sz == 0) {
-		/* Meta blob not specified */
+			rc = stat(path, &lstat);
+			if (rc != 0)
+				return daos_errno2der(errno);
+			meta_sz = lstat.st_size;
+		}
+		/* If scm_sz is zero and meta_sz is not, use input meta_sz */
+	} else {
```

Comment:
From my understanding, if a custom `meta_sz` is given, it should not be overloaded with `scm_sz` ?
Thus the old `} else if (meta_sz == 0) {` seems to still be relevant from my understanding.

Reply:
@NiuYawei pointed out on a previous iteration that there are multiple use cases for vos_pool_create() other than standard DAOS pool creation (e.g. RDB/SysDB/tests) that use a different method of creation (automatic/deferred allocation rather than pre-allocation). This intent is signalled through the scm_sz parameter which is set to 0 for pre-allocation and non-zero for deferred allocation. In order to preserve this mechanism, meta_sz input parameter is only used when scm_sz is 0 which indicates the standard DAOS pool creation case. That is how I understand it, please correct me if I'm missing something.

Reply:
Thanks for the explanation, I have to admit that I did not make the link with Niu answer.
Seems logical from my understanding.